From fbb9dc6e291eaa058f3ef1cd93711265bfbfd162 Mon Sep 17 00:00:00 2001
From: Heiko Becker <heirecka@exherbo.org>
Date: Fri, 7 Nov 2014 00:11:02 +0100
Subject: [PATCH] Workaround build failures with boost>=1.57 and moc

Otherwise I get the following error message:
'usr/include/boost/type_traits/detail/has_binary_operator.hp:50:
Parse error at "BOOST_JOIN"'. See
https://bugreports.qt-project.org/browse/QTBUG-22829 for details.

The old workaround to define BOOST_TT_HAS_OPERATOR_HPP_INCLUDED
doesn't seem to work here.
---
 src/QGlib/connect.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/QGlib/connect.cpp b/src/QGlib/connect.cpp
index 9ff6fde..58a3367 100644
--- a/src/QGlib/connect.cpp
+++ b/src/QGlib/connect.cpp
@@ -21,8 +21,10 @@
 #include <QtCore/QHash>
 #include <QtCore/QMutex>
 #include <boost/multi_index_container.hpp>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/multi_index/sequenced_index.hpp>
 #include <boost/multi_index/ordered_index.hpp>
+#endif
 #include <boost/multi_index/member.hpp>
 
 namespace QGlib {
-- 
2.1.3

